netty/src/main/java/io/grpc/netty/NettyClientStream.java

```diff
@@ -109,6 +110,11 @@ public void operationComplete(ChannelFuture future) throws Exception {
         !method.getType().clientSendsOneMessage()).addListener(failureListener);
   }
 
+  /**
+   * Intended to be overriden by NettyClientTransport, which has more information about failures.
+   */
+  protected abstract Status statusFromThrowableFromPromise(Throwable t);
```

Comment:
Nit: this name is a little confusing ... would something like `statusFromFailedPromise` capture the intent? Also, would it make sense for the parameter to be `ChannelFuture`?